src/matlab2tikz.m

```diff
@@ -1048,17 +1048,25 @@ case guitypes()
     % Get legend handle associated with current axis
 
     legendhandle = [];
-    env = getEnvironment();
+    [env, envVersion] = getEnvironment();
     switch env
         case 'Octave'
             % Make sure that m2t.legendHandles is a row vector.
             for lhandle = m2t.legendHandles(:)'
-                ud = get(lhandle, 'UserData');
+                if isVersionBelow(envVersion, [4,2,2]) % 5865d2fef424
+                  lhandleProp{1}='UserData';
+                  lhandleProp{2}='handle';
+                else
+                  lhandleProp{1}='__appdata__';
+                  lhandleProp{2}='__axes_handle__';
+                end
+                ud = get(lhandle, lhandleProp{1});
                 % Empty if no legend and multiple handles if plotyy
-                if ~isempty(ud) && any(axisHandle == ud.handle)
+                if ~isempty(ud) && any(axisHandle == ud.(lhandleProp{2}))
                     legendhandle = lhandle;
                     break
                 end
+                clear lhandleProp
```

Comment:
```suggestion
```

Clear is not necessary. Yes, we have more variables in the local workspace, but I have also noticed that it can slow down a function considerably. So I would just remove it, unless it clearly interferes with different code (and in that case, it would be better to pick a different name for that variable.